apis/python/src/tiledbsoma/tiledb_object.py

```diff
@@ -49,6 +53,29 @@ def __init__(
 
         self.metadata = MetadataMapping(self)
 
+    def _default_ctx(self) -> tiledb.Ctx:
+        """
+        The TileDB context used when no other is supplied. Must have good defaults for positive
+        out-of-the-box UX.
+        """
+
+        cfg = {}
+
+        # Crucial for reducing number of HTTP requests for tiledb-cloud URIs.
+        # With default (small) value, it can take many HTTP-request round trips to satisfy
+        # a single query.
+        cfg["py.init_buffer_bytes"] = 4 * 1024**3
+
+        # This is necessary for smaller tile capacities when querying with a smaller memory budget.
+        cfg["sm.mem.reader.sparse_global_order.ratio_array_data"] = 0.3  # type: ignore
+
+        # Temp workaround pending https://app.shortcut.com/tiledb-inc/story/23827
+        region = os.getenv("AWS_DEFAULT_REGION")
```

Comment:
very happy to have this change!  however, this only fixes it when the  user is not providing a configu/ctx (i.e., it sets it in the default context, but not otherwise).

Ideally, it would "honor" the environment variable in any situation where the context does not have "vfs.s3.region" set, and the environment variable is set...    Is such a thing possible, or does it need to wait for the real fix?

In the case of the cell census, we _always_ provide a tiledb context when opening the census, so the above fix doesn't help us....

Reply:
> Is such a thing possible, or does it need to wait for the real fix?

Yes, very much doable -- good feedback, thanks!!

Reply:
... oops, correction to what I said, @bkmartinjr -- even if you make a `ctx = tiledb.Ctx({})` totally plain-vanilla and _you_ (the caller) aren't setting `vfs.s3.region` ... still `ctx.Config()['vfs.s3.region']` _does_ have `us-east-1` set as the default. (Which is the core bug.) So at the spot on this PR we can't find out if the caller explicitly specified `vfs.s3.region` or not. :(

So what I think we need to do for this PR -- temporarily, for now, only pending TileDB-internal tracking task 23827, and only for soma -- is maybe _always_ use `AWS_DEFAULT_REGION` if it exists in the env -- ? What do you think?

cc @ihnorton

Reply:
Also if we don't have closure today, I'd like to merge this PR as-is -- as it is an improvement on the out-of-the-box UX which will help uptake -- and try to go farther on a follow-on PR.

Reply:
It seems like there are edge cases either way - I would rather not have the environment variable override the explicitly set context (that would screw up the cell-census package in a big way ,as our bucket is in us-west-2, and who knows what value the user will have their environment variable set too).

My pref:  land this as is.